hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/queries/AsNodeOperatorQueriesTest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.services.bdd.suites.queries;
+
+import static com.hedera.services.bdd.junit.TestTags.CRYPTO;
+import static com.hedera.services.bdd.spec.HapiSpec.hapiTest;
+import static com.hedera.services.bdd.spec.queries.QueryVerbs.getAccountBalance;
+import static com.hedera.services.bdd.spec.queries.QueryVerbs.getAccountInfo;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.cryptoCreate;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.doAdhoc;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.sleepFor;
+import static com.hedera.services.bdd.suites.HapiSuite.ONE_HUNDRED_HBARS;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.hedera.services.bdd.junit.HapiTest;
+import com.hedera.services.bdd.junit.HapiTestLifecycle;
+import com.hedera.services.bdd.junit.support.TestLifecycle;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Tag;
+
+@Tag(CRYPTO)
+@HapiTestLifecycle
+@DisplayName("Node Operator Queries")
+/**
+ * A class with Node Operator Queries tests
+ */
+public class AsNodeOperatorQueriesTest extends NodeOperatorQueriesBase {
+
+    @Nested
+    @DisplayName("verify payer balance")
+    /**
+     * A class with Node Operator tests that verify the payer balance
+     */
+    class PerformNodeOperatorQueryAndVerifyPayerBalance {
+
+        @BeforeAll
+        static void beforeAll(@NonNull final TestLifecycle lifecycle) {
+            lifecycle.doAdhoc(createAllAccountsAndTokens());
+        }
+
+        @HapiTest
+        final Stream<DynamicTest> nodeOperatorQueryVerifyPayerBalanceForAccountBalance_WithoutOpContext() {
+            // declare payer account balance variables
+            final AtomicLong currentPayerBalance = new AtomicLong();
+            final AtomicLong payerBalanceAfterFirstQuery = new AtomicLong();
+            final AtomicLong payerBalanceAfterSecondQuery = new AtomicLong();
```

Comment:
I still think the slight flexibility the atomic vars approach offers isn't worth the cost, for the following reasons: 

- We can't actually see an expected amount anywhere in the test, meaning other factors (e.g. a variable exchange rate, or any other network-wide property) could influence the balances and we wouldn't necessarily know, potentially making debugging harder 
- It's awkward to store every query result (e.g. `payerBalanceAfterFirstQuery`...what if there are 18 queries?) We're essentially getting around the intended use of the framework by not using known verbs like `has(accountWith().balance(123L))`
- This is significantly different than how the vast majority of hapi tests are written 

BUT, if this is how the team wants to write tests going forward, I'll be quiet 😂 If that is the case, we should add a ticket to make these types of assertions more 'supported' by the framework.

Reply:
Matt, I understand your concerns, here is how I can change the tests (still not committed):

// the tests but without the atomic long variables
        @HapiTest
        final Stream<DynamicTest> nodeOperatorQueryVerifyPayerBalanceForAccountBalance_WithoutAtomicLong() {
            return hapiTest(
                    cryptoCreate(NODE_OPERATOR).balance(ONE_HUNDRED_HBARS),
                    cryptoCreate(PAYER).balance(ONE_HUNDRED_HBARS),
                    // perform getAccountBalance() query, pay for the query with payer account
                    // the grpc client performs the query to different ports
                    getAccountBalance(NODE_OPERATOR).payingWith(PAYER),
                    sleepFor(3_000),
                    // assert payer is charged
                    getAccountBalance(PAYER).hasTinyBars(ONE_HUNDRED_HBARS),
                    // perform free query to local port with asNodeOperator() method
                    getAccountBalance(NODE_OPERATOR).payingWith(PAYER).asNodeOperator(),
                    sleepFor(3_000),
                    // assert payer is not charged as the query is performed as node operator
                    getAccountBalance(PAYER).hasTinyBars(ONE_HUNDRED_HBARS));
        }

        @HapiTest
        final Stream<DynamicTest> nodeOperatorQueryVerifyPayerBalanceForAccountInfo_WithoutAtomicLong() {
            return hapiTest(
                    cryptoCreate(NODE_OPERATOR).balance(ONE_HUNDRED_HBARS),
                    cryptoCreate(PAYER).balance(ONE_HUNDRED_HBARS),
                    // perform getAccountInfo() query, pay for the query with payer account
                    // the grpc client performs the query to different ports
                    getAccountInfo(NODE_OPERATOR).payingWith(PAYER),
                    sleepFor(3_000),
                    // assert payer is charged
                    getAccountBalance(PAYER).hasTinyBars(ONE_HUNDRED_HBARS - QUERY_COST),
                    // perform free query to local port with asNodeOperator() method
                    getAccountInfo(NODE_OPERATOR).payingWith(PAYER).asNodeOperator(),
                    sleepFor(3_000),
                    // assert payer is not charged as the query is performed as node operator
                    getAccountBalance(PAYER).hasTinyBars(ONE_HUNDRED_HBARS - QUERY_COST));
        }

   For the first test there is no headache as the query is free of charge in general, but what bothers me is the case when we will be charged for a query. What I understand is that we will have to know the exact amount that will be charged for the query. Can we rely that this amount will be always the same and the tests won't fail for example after some release, because these charges were changed? If we are sure that the amounts are always the same, then it is ok, but if we are not sure, then the risk here will be that potentially these tests might start failing because we are asserting concrete amounts and they will require some maintenance in the future. 
   That's what was bothering me here from the beginning and this is why I created the tests with the variables, I was thinking for some kind of flexibility but of course I will change them right away if you think that is better, that was just my vision at the beginning.

Reply:
Query and transaction costs don't change very much, generally speaking. I've never implemented a cost change, but I imagine when we do, we revisit all of the tests that fail to make sure they are still relevant and set up correctly. 

But guess what? I found a test that does something similar [here](https://github.com/hashgraph/hedera-services/blob/2c30a3ac0ddb31a4f2289b33ddc8ac0c20f5bb70/hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java#L2080)! It uses the `balanceSnapshot(...)` and `changeFromSnapshot(...)` methods to compute expected deltas without the atomic vars 😄 What do you think of something like this?

```java
    @HapiTest
    final Stream<DynamicTest> nodeOperatorQueryVerifyPayerBalanceForAccountInfo_WithoutAtomicLong() {
        return hapiTest(
                cryptoCreate(NODE_OPERATOR).balance(ONE_HUNDRED_HBARS),
                cryptoCreate(PAYER).balance(ONE_HUNDRED_HBARS),
                balanceSnapshot("payerInitialBalance", PAYER),
                // perform getAccountInfo() query, pay for the query with payer account
                // the grpc client performs the query to different ports
                getAccountInfo(NODE_OPERATOR).payingWith(PAYER),
                sleepFor(3_000),
                // assert payer is charged
                getAccountBalance(PAYER).hasTinyBars(changeFromSnapshot("payerInitial", -QUERY_COST)),
                // perform free query to local port with asNodeOperator() method
                getAccountInfo(NODE_OPERATOR).payingWith(PAYER).asNodeOperator(),
                sleepFor(3_000),
                // assert payer is not charged as the query is performed as node operator
                getAccountBalance(PAYER).hasTinyBars(changeFromSnapshot("payerInitial", -QUERY_COST)));
    }
```

Reply:
I didn't realize that node operator queries were a high priority, so we don't need to spend lots of time on this now. Let's get this PR merged, then we can come back to this.